cmd/server/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"google.golang.org/grpc"
+
+	"github.com/Pratham700/github-search-service/internal/server"
+	pb "github.com/Pratham700/github-search-service/proto/proto"
+)
+
+func main() {
+	// Set the port for the gRPC server
+	port := flag.Int("port", 50051, "The port number to listen on")
+	flag.Parse() // Parse the command-line flags
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", *port))
+	if err != nil {
+		log.Fatalf("failed to listen: %v", err)
+	}
+
+	// Create a new gRPC server
+	s := grpc.NewServer(grpc.UnaryInterceptor(server.AuthInterceptor))
```

Comment:
You don't need the type assert here. The Go compiler can figure it out.

```suggestion
	s := grpc.NewServer(server.AuthInterceptor)
```

Reply:
That's interesting.

Reply:
I am not sure the IDE is complaining by having that suggestion.